Modules/getpath.py

```diff
@@ -593,6 +593,13 @@ def search_up(prefix, *landmarks, test=isfile):
         if prefix and not stdlib_dir:
             stdlib_dir = joinpath(prefix, STDLIB_SUBDIR)
 
+    # Note: the `home` variable in pyvenv.cfg is not always accurate.
+    # Detect prefix by searching from *real* executable location for the stdlib_dir.
+    if STDLIB_SUBDIR and STDLIB_LANDMARKS and base_executable and not prefix:
+        prefix = search_up(base_executable, *STDLIB_LANDMARKS)
+        if prefix and not stdlib_dir:
+            stdlib_dir = joinpath(prefix, STDLIB_SUBDIR)
```

Comment:
Let's make this a bit more intentional. My suggestion:

- Set a `pyvenvcfg_home` variable when parsing `pyvenv.cfg`
- Move this search block to after the `PREFIX and not prefix` check
- Change the check to `pyvenvcfg_home and STDLIB_SUBDIR and STDLIB_LANDMARKS and base_executable and not prefix`
- Reference `GH-127440` in the comment

Reply:
> Move this search block to after the PREFIX and not prefix check

It seems that we should not move this block, because 
```python
    if PREFIX and not prefix:
        prefix = PREFIX
        ...
``` 
will set the `prefix`, which will cause `pyvenvcfg_home and STDLIB_SUBDIR and STDLIB_LANDMARKS and base_executable and not prefix` check to fail.

If you mean to modify ` if PREFIX and not prefix:...` to `if pyvenvcfg_home and STDLIB_SUBDIR and STDLIB_LANDMARKS and base_executable and not prefix`, maybe we need to modify more test cases.

Reply:
Right, it needs some adjusting, but I think we should still fallback to `PREFIX` before `base_executable`.

```py
if PREFIX and STDLIB_LANDMARKS and not prefix and any(isfile(joinpath(PREFIX, f)) for f in STDLIB_LANDMARKS):
    prefix = PREFIX

if pyvenvcfg_home and STDLIB_SUBDIR and STDLIB_LANDMARKS and base_executable and not prefix:
    ...

if not prefix:
    prefix = abspath('')
    warn('Could not find platform independent libraries <prefix>')
```